encode.go

```diff
@@ -406,7 +408,12 @@ func (e *Encoder) isNeedQuoted(v string) bool {
 
 func (e *Encoder) encodeString(v string, column int) ast.Node {
 	if e.isNeedQuoted(v) {
-		v = strconv.Quote(v)
+		switch e.quoteStyle {
+		case QuoteStyleSingle:
+			v = quoteWith(v, '\'')
+		default:
+			v = strconv.Quote(v)
```

Comment:
This is pretty much the same as calling quoteWith now. Should we just call quoteWith?